### DISCUSSION TITLE
Returning FileNotFoundError and allowing relative filepaths

### DISCUSSION THREAD

**Discussion by andibarg** [MREYE-LUMC/ZOSPy#34] [Ideas]
Hi,

I realized the `load` function in `OpticStudioSystem` does not return any errors when no file is found. This got me confused recently as I was trying to figure out why a zospy analysis function returned nonsense and my script did not complain. It took me a while to find that I had misspelled the filename in `load` all the way in the beginning of my script :).

Do you agree that there should be a FileNotFoundError or similar raised?

On a related note, I was wondering if we should provide the possibility for relative filepaths. I image we could add the following to `load`:

```python
if os.path.dirname(filepath) == '':
    filepath = os.path.join(os.getcwd(),filepath)
``` 
What do you think?

**Comment by crnh** [comment]
I've been thinking about this as well and envisioned it as part of a broader refactoring of zpcore. I'd like to add 2 points to the discussion:

1. If we are going to do anything with paths, we should consider using pathlib instead of fiddling with strings ourselves. Strings are not the most reliable way to represent paths.
2. This discussion is also relevant for the `save` and `save_as` methods.

The requested functionality can be implemented using pathlib's `resolve` method, which converts relative paths to absolute paths and can check for their existence.

**Comment by crnh** [comment]
This has been implemented in #62 and just released with ZOSPy 1.2.0!
